src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AspNetCore/GenericHostWebHostBuilderExtensionsWrapper.cs

```diff
@@ -32,20 +33,24 @@ private static class AspNetCore21Types
         {
             public static void AddStartupFilterToHostBuilder(object hostBuilder, IAgent agent)
             {
-                var typedHostBuilder = (Microsoft.Extensions.Hosting.IHostBuilder)hostBuilder;
-                typedHostBuilder.ConfigureServices(AddStartupFilter);
-
-                void AddStartupFilter(Microsoft.Extensions.Hosting.HostBuilderContext hostBuilderContext, IServiceCollection services)
+                if (!_isNewRelicStartupFilterAdded)
```

Comment:
After thinking about this more, I'd prefer to treat this block of code as a critical section. If we did something like:
```
if (0 == Interlocked.CompareExchange(ref _isNewRelicStartupFilterAdded, 1, 0))
```
That should help ensure that even in a concurrent scenario we will only ever call `ConfigureServices` once.